kmath-core/src/commonMain/kotlin/space/kscience/kmath/structures/Buffer.kt

```diff
@@ -100,9 +100,29 @@ public fun <T> Buffer<T>.asSequence(): Sequence<T> = Sequence(::iterator)
 public fun <T> Buffer<T>.asIterable(): Iterable<T> = Iterable(::iterator)
 
 /**
- * Converts this [Buffer] to a new [List]
+ * Returns a new [List] containing all elements of this buffer.
  */
-public fun <T> Buffer<T>.toList(): List<T> = asSequence().toList()
+public fun <T> Buffer<T>.toList(): List<T> = when (this) {
+    is ArrayBuffer<T> -> array.toList()
+    is ListBuffer<T> -> list.toList()
+    is MutableListBuffer<T> -> list.toList()
+    else -> asSequence().toList()
+}
+
+/**
+ * Returns a new [MutableList] filled with all elements of this buffer.
+ */
+public fun <T> Buffer<T>.toMutableList(): MutableList<T> = when (this) {
```

Comment:
It should be a MuttableBuffer

Reply:
One should be able to create a mutable copy of an immutable collection instance, e.g. : listOf(1).toMutableList()

Reply:
It does not make any sense. If he wants to create a mutable copy, he can do `toMutableList`, but I can't see any applications to that.